docs/types.rst

```diff
@@ -337,7 +337,8 @@ function type should not return anything, the whole ``returns (<return types>)``
 part has to be omitted.
 
 By default, function types are internal, so the ``internal`` keyword can be
-omitted.
+omitted. Note that this is not the case for functions defined on a contract,
```

Comment:
I do not think this is the right way to explain. If you look up a few lines, that lists what is valid for a function type. This would weirdly suggest `public` is also valid a function type.

Having read the entire section again I think the intro should be clarified, otherwise this feels like only a workaround.